src/NumericalAlgorithms/LinearOperators/PowerMonitors.cpp

```diff
@@ -56,6 +89,65 @@ std::array<DataVector, Dim> power_monitors(
   return result;
 }
 
+template <size_t Dim>
+void relative_truncation_error(
+    const gsl::not_null<std::array<double, Dim>*> result,
+    const DataVector& input_data_vector, const Mesh<Dim>& mesh) {
+  // Temporary: compute power monitors
+  const auto pm_array = power_monitors<Dim>(input_data_vector, mesh);
+
+  // Compute relative truncation error in each dimension
+  size_t number_of_points_per_dim = 0;
+  for (size_t sliced_dim = 0; sliced_dim < Dim; ++sliced_dim) {
+    // Number of power monitors per dimension
+    number_of_points_per_dim = gsl::at(pm_array, sliced_dim).size();
```

Comment:
(optional) I don't think that moving the `size_t` out of the loop makes a big difference in performance (or any at all? not sure). I'd do this:

```cpp
const auto& modes =  gsl::at(pm_array, d);
detail::rel_trunc_err(modes, modes.size());
```